SIL.Windows.Forms.Keyboarding/Linux/CombinedIbusKeyboardRetrievingAdaptor.cs

```diff
@@ -146,20 +155,54 @@ private static void LoadDefaultXkbSettings()
 		/// </summary>
 		private static void LoadLatinLayouts(IntPtr settingsGeneral)
 		{
-			IntPtr value = Unmanaged.g_settings_get_value(settingsGeneral, "xkb-latin-layouts");
-			CombinedIbusKeyboardSwitchingAdaptor.LatinLayouts =
-				KeyboardRetrievingHelper.GetStringArrayFromGVariantArray(value);
-			Unmanaged.g_variant_unref(value);
 
-			CombinedIbusKeyboardSwitchingAdaptor.UseXmodmap = Unmanaged.g_settings_get_boolean(
-				settingsGeneral, "use-xmodmap");
+			CombinedIbusKeyboardSwitchingAdaptor.LatinLayouts = GetXkbLatinLayouts(settingsGeneral);
+			string useXModmapKey = "use-xmodmap";
+			bool useXModmap = false;
+			if (Platform.IsFlatpak)
+				useXModmap = GSettingsGetBooleanFromHost(GSettingsSchemaId, useXModmapKey);
+			else
+				useXModmap = Unmanaged.g_settings_get_boolean(settingsGeneral, useXModmapKey);
+			CombinedIbusKeyboardSwitchingAdaptor.UseXmodmap = useXModmap;
 			//Console.WriteLine("DEBUG use-xmodmap = {0}", _use_xmodmap);
 			//Console.Write("DEBUG xkb-latin-layouts =");
 			//for (int i = 0; i < _latinLayouts.Length; ++i)
 			//	Console.Write("  '{0}'", _latinLayouts[i]);
 			//Console.WriteLine();
 		}
 
+		private static string[] GetXkbLatinLayouts(IntPtr settingsGeneral)
+		{
+			string keyName = "xkb-latin-layouts";
+			if (Platform.IsFlatpak)
+			{
+				string output = KeyboardRetrievingHelper.RunOnHostEvenIfFlatpak("gsettings", $"get {GSettingsSchemaId} {keyName}").StandardOutput;
```

Comment:
To make this more similar to a few lines above, you could refactor this line into a trivial method `GSettingsGetStringFromHost`. As it is now I found the two method names `GSettingsGetBooleanFromHost` vs `RunOnHostEvenIfFlatpak` a bit confusing (until I looked how `GSettingsGetBooleanFromHost` is implemented).

Reply:
Done.